parser/src/parsercontext.cpp

```diff
@@ -0,0 +1,81 @@
+#include <boost/filesystem.hpp>
+
+#include <model/file.h>
+#include <model/file-odb.hxx>
+
+#include <util/hash.h>
+#include <util/odbtransaction.h>
+
+#include <parser/parsercontext.h>
+#include <parser/sourcemanager.h>
+
+namespace po = boost::program_options;
+
+namespace cc
+{
+namespace parser
+{
+
+ParserContext::ParserContext(
+    std::shared_ptr<odb::database> db_,
+    SourceManager& srcMgr_,
+    std::string& compassRoot_,
+    po::variables_map& options_):
+      db(db_),
+      srcMgr(srcMgr_),
+      options(options_),
+      compassRoot(compassRoot_)
+{
+  std::unordered_map<std::string, std::string> fileHashes;
+
+  (util::OdbTransaction(this->db))([&]
+   {
+     // Fetch all files from SourceManager
```

Comment:
.

Reply:
Guess you are not joking. Still should not be a change request at the very first place.